ktor-server/ktor-server-tests/jvm/test/io/ktor/server/plugins/StaticContentTest.kt

```diff
@@ -1273,6 +1275,56 @@ class StaticContentTest {
             }
         }
     }
+
+    @Test
+    fun testStaticPathFromChangingZip() = testApplication {
+        val path = "jvm/test-resources/dynamic.zip"
+        val firstFileName = "firstFile.txt"
+        val secondFileName = "secondFile.txt"
+        val firstContent = "Hello"
+        val secondContent = "World"
+        val firstZipFile = createZipFile(path, firstFileName, firstContent)
+
+        routing {
+            staticZip(
+                remotePath = "static",
+                basePath = null,
+                zip = Paths.get(path),
+                reloadOnRequest = true
+            )
+        }
+
+        val firstResponse = client.get("static/$firstFileName")
+        assertEquals(HttpStatusCode.OK, firstResponse.status)
+        assertEquals(firstContent, firstResponse.bodyAsText())
+
+        firstZipFile.delete()
+        val secondZipFile = createZipFile(path, secondFileName, secondContent)
+
+        // there is a delay in the file system watcher about 2 seconds
+        delay(3000)
```

Comment:
Maybe to avoid flakiness here, you could introduce another file system watcher in the test with a mutex to block on, so when the test FS watcher triggers we can assume that the zip watcher has triggered.

Reply:
Good idea, thanks

Reply:
Ah, I was hoping the delay could be removed entirely, I didn't realize it was 2 second polling behind the scenes.  Would be nice if we could avoid introducing long tests like this, but I don't see any way around it here.